cluster-autoscaler/proposals/metrics.md

```diff
@@ -0,0 +1,98 @@
+## Cluster Autoscaler Monitoring
+
+## Introduction
+Currently the options to monitor Cluster Autoscaler are limited to logs, status
+configmap and events. While useful for debugging, none of this options is
+paritcularly practical for monitoring Cluster Autoscaler behaviour over time.
+This document describes a set of metrics (in Prometheus format) that will be
+added to 0.6 Cluster Autoscaler to allow better monitoring of its operations.
+
+## Current state
+Cluster Autoscaler already has a metrics endpoint providing some basic metrics.
+This includes default process metrics (number of goroutines, gc duration, cpu
+and memory details, etc) as well as some custom metrics related to time taken by
+various parts of Cluster Autoscaler main loop. Existing metrics will be renamed
+and refactored and new metrics will be added.
+
+## Metrics
+Cluster Autoscaler needs to monitor some aspects of cluster state as part of its normal operations. It can therefore provide metrics describing its own operations as well as some metrics related to general state of the cluster.
+
+All the metrics are prefixed with `cluster_autoscaler_`.
+
+### Cluster state
+
+| Metric name | Metric type | Labels | Description |
+| ----------- | ----------- | ------ | ----------- |
+| cluster_healthy | Gauge | | Whether or not cluster is healthy enough for autoscaling. 1 if it is, 0 otherwise. |
```

Comment:
The name of this metric indicates, that this is a general cluster health metric, but as far as I understand it's quite specific to how the autoscaler assumes what is safe to autoscale. Therefore I'd rename this to `cluster_safe_to_autoscale`.